contracts/libraries/MultiSendCallOnly.sol

```diff
@@ -59,8 +59,9 @@ contract MultiSendCallOnly {
                 }
                 if eq(success, 0) {
                     let errorLength := returndatasize()
-                    returndatacopy(0, 0, errorLength)
-                    revert(0, errorLength)
+                    let returnPtr := mload(0x40)
```

Comment:
I'd check with the team about the real motivation behind this ticket: e.g., do we want to fix the compilation error or make the contracts fully memory safe? AFAIK the multisend contract compiled just fine with the optimizer

Reply:
The multisend contract is not memory safe as-is though - meaning that if we included it in the the compilation unit (which `hardhat` does - it includes all contracts to compile with a single compiler version in the same compilation unit), then the contracts as a whole will not be considered memory safe (and allow IR optimizations). At least that is my understanding of how things work.

The title of the issue is:

> Enabling yul optimizer and viaIR causes stack too deep error using Solidity 0.8.19

So, at least AFAIU the issue, we want to make it so IR optimization work, and this requires that all contracts be memory safe, and AFAIU `MultiSend` is not.

Reply:
From the solc maintainer: https://github.com/safe-global/safe-contracts/issues/544#issuecomment-1498952802

> All blocks that are included in the code a contract is built from. So e.g. if you have two different contracts in a file and they don't inherit from each other, blocks in one do not influence the other.

Reply:
Ahhhhhh, OK. thanks for correcting my understanding. I thought it was any contract in the _compilation unit_ and not just any contract in the _dependency hierarchy_.

In that case, I'm all for **not** changing the `MultiSend*` and `DelegateCaller` contracts.

Reply:
Done 👍🏾